dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-reports-reconciliation-summary.controller/helpers.ts

```diff
@@ -17,29 +17,32 @@ import { isEqualMonthAndYear } from '../../../../utils/date';
 
 type UtilisationReportForSubmissionMonth = {
   submissionMonth: IsoMonthStamp;
-  report: UtilisationReport;
+  report: UtilisationReportEntity;
 };
 
 type SummaryItemForSubmissionMonth = {
   submissionMonth: IsoMonthStamp;
   item: UtilisationReportReconciliationSummaryItem;
 };
 
-const mapToSummaryItem = async (bank: Bank, report: UtilisationReport): Promise<UtilisationReportReconciliationSummaryItem> => {
-  const reportData = await getAllUtilisationDataForReport(report);
+const mapToSummaryItem = async (bank: Bank, report: UtilisationReportEntity): Promise<UtilisationReportReconciliationSummaryItem> => {
+  const reportData = await UtilisationDataRepo.findByReport(report);
+  if (!reportData) {
```

Comment:
Is it possible for this to return a falsy value? My guess would be this either throws an error ( e.g. because it can't get connect to the db) or returns an array, which are always truthy in javascript.

Reply:
This was a mistake from copying other stuff I think. The return type I put for `.findByReport` was `Promise<UtilisationDataEntity[] | null>`, but it will always be `Promise<UtilisationDataEntity[]>`, so the thing that we would actually want to check for is `reportData.length` being zero. However, it's fine for there to be no report data, so this check should be removed entirely. Thanks for pointing it out!